plugins/modules/filesystem.py

```diff
@@ -1363,8 +1366,10 @@ def is_owner_modified(self, effective_path, owner):
                 zone=self.module.params['access_zone'],
                 provider=owner_provider)
 
-            owner_uid = owner_details['users'][0]['uid']['id']
-            owner_sid = owner_details['users'][0]['sid']['id']
+            # use sid for user in ads provider
+            user = owner_details['users'][0]
+            owner_uid = user['uid']['id'] if owner_provider != 'ads' else user['sid']['id']
```

Comment:
I feel we can skip the if-else statement here
```suggestion
            owner_uid = user['uid']['id']
```
because we already have a check based on provider https://github.com/dell/ansible-powerscale/pull/177/files#diff-1462c76fc9abe665cfa5859ddf27bab518978ba1d199d98218b5b736a60f6f9bR1389

Reply:
Hi @Bhavneet-Sharma , we do have check below, but we still need to ensure the object @Line1374
```
owner = {'type': 'user', 'id': owner_uid,
                     'name': owner['name']}
```
has the correct uid/sid and apply the modification.